test-loop-tests/src/setup/builder.rs

```diff
@@ -333,18 +334,18 @@ impl<'a> NodeStateBuilder<'a> {
             config_modifier(&mut client_config);
         }
 
-        NodeSetupState { account_id, client_config, store, split_store }
+        NodeSetupState { account_id, client_config, store, split_store, cold_db }
     }
 
-    fn setup_store(&self) -> (Store, Option<Store>) {
-        let (store, split_store) = if self.archive {
-            let (hot_store, split_store) = create_test_split_store();
-            (hot_store, Some(split_store))
+    fn setup_store(&self) -> (Store, Option<Store>, Option<Arc<ColdDB>>) {
+        let (store, split_store, cold_db) = if self.archive {
+            let (hot_store, split_store, cold_db) = create_test_split_store();
```

Comment:
nit: Another option is to create a `TestStore` or similarly named struct with `Store, Option<Store>, Option<Arc<ColdDB>>` and make that the return type of both `setup_store` and `create_test_split_store`